crates/sui-framework/sources/coin.move

```diff
@@ -234,6 +271,43 @@ module sui::coin {
         burn(c, coin);
     }
 
+    // === Update coin metadata ===
+
+    /// Update decimals in `CoinMetadata`
+    public entry fun update_decimals<T>(
+        _treasury: &TreasuryCap<T>, metadata: &mut CoinMetadata<T>, decimals: u8
+    ) {
+        metadata.decimals = decimals;
+    }
+
+    /// Update name of the coin in `CoinMetadata`
+    public entry fun update_name<T>(
```

Comment:
Should a coin be allowed to change its name without becoming a different currency?

Reply:
IMO yes. Name is just a display name. Classic example is GOOG stock is alphabet.